iconservice/icon_service_engine.py

```diff
@@ -441,6 +444,105 @@ def _invoke_request(self,
 
         return self._call(context, method, params)
 
+    def _estimate_step_by_request(self, request, context):
+        """Calculates simply and estimates step with request data.
+
+        :param request:
+        :param context:
+        """
+        params: dict = request['params']
+        data_type: str = params.get('dataType')
+        data: dict = params.get('data')
+        to: Address = params['to']
+
+        context.step_counter.apply_step(StepType.DEFAULT, 1)
+
+        input_size = self._get_byte_length(params.get('data', None))
+        context.step_counter.apply_step(StepType.INPUT, input_size)
+
+        if data_type == "deploy":
+            data_size = self._get_byte_length(data.get('content', None))
+            context.step_counter.apply_step(StepType.CONTRACT_SET, data_size)
+            # When installing SCORE.
+            if to == ZERO_SCORE_ADDRESS:
+                context.step_counter.apply_step(StepType.CONTRACT_CREATE, 1)
+            # When updating SCORE.
+            else:
+                context.step_counter.apply_step(StepType.CONTRACT_UPDATE, 1)
+
+    def _estimate_step_by_execution(self, request, context, step_limit):
+        """Processes the transaction and estimates step.
+
+        :param request:
+        :param context:
+        :param step_limit:
+        """
+        method: str = request['method']
+        params: dict = request['params']
+
+        from_: Address = params['from']
+        to: Address = params['to']
+
+        timestamp = params.get('timestamp', self._icx_storage.last_block.timestamp)
+        context.tx = Transaction(tx_hash=sha3_256(int_to_bytes(timestamp)),
```

Comment:
```suggestion
        context.tx = Transaction(tx_hash=sha3_256(int_to_bytes(timestamp)).digest(),
```

Reply:
def sha3_256(data: bytes) -> bytes:
    return hashlib.sha3_256(data).digest() 

I used the function at util. It already did to digest data.